ci/redmine_bugzilla.py

```diff
@@ -132,22 +142,39 @@ def main():
                             if external_bug['ext_priority'] != issue.priority.name:
                                 ext_params['ext_priority'] = issue.priority.name
                             if len(ext_params.keys()) > 0:
-                                ext_bug_record += 'Bugzilla bug %s updated from upstream bug %s with ' \
-                                                  '%s\n' % (bug.id, issue.id, ext_params)
+                                ext_bug_record += 'Bugzilla bug %s updated from upstream bug %s ' \
+                                                  'with %s\n' % (bug.id, issue.id, ext_params)
                                 ext_params['ids'] = external_bug['id']
                                 BZ.update_external_tracker(**ext_params)
                                 if 'ext_status' in ext_params:
                                     bug.addcomment(
-                                        'The Pulp upstream bug status is at %s. Updating the external '
-                                        'tracker on this bug.' % issue.status.name)
+                                        'The Pulp upstream bug status is at %s. Updating the '
+                                        'external tracker on this bug.' % issue.status.name)
                                 if 'ext_priority' in ext_params:
                                     bug.addcomment(
                                         'The Pulp upstream bug priority is at %s. Updating the '
                                         'external tracker on this bug.' % issue.priority.name)
+                            if bug.status in ['NEW', 'ASSIGNED']:
+                                if issue.status.name in ['MODIFIED', 'ON_QA', 'VERIFIED',
+                                                         'CLOSED - CURRENTRELEASE']:
+                                    transition_to_post.append(True)
+                                else:
+                                    transition_to_post.append(False)
+                            downstream_POST_plus = ['POST', 'MODIFIED', 'ON_QA', 'VERIFIED',
+                                                    'RELEASE_PENDING', 'CLOSED']
+                            upstream_POST_minus = ['NEW', 'ASSIGNED', 'POST']
+                            if bug.status in downstream_POST_plus and \
+                                            issue.status.name in upstream_POST_minus:
+                                msg = 'The downstream bug %s is at POST+ but the upstream bug %s ' \
+                                      'at POST-. Something is wrong.' % (bug.id, issue.id)
+                                raise RuntimeError(msg)
```

Comment:
you should probably accumulate these error messages like you do with the linking problems. Then raise an Exception at the end with text from all the problems found along the way.